internal/agent/device/systeminfo/system_info.go

```diff
@@ -564,6 +564,16 @@ func getSystemInfoMap(ctx context.Context, log *log.PrefixLogger, info *Info, in
 		infoMap[key] = val
 	}
 
+	for key, collectorfn := range collectors {
+		_, alreadyExists := infoMap[key]
+		if alreadyExists {
+			log.Debugf("SystemInfo collector already populated: %s is %s", key, infoMap[key])
```

Comment:
nit: we should really warn here as this would be unexpected to duplicate an internal implementation.